src/io/quic.rs

```diff
@@ -0,0 +1,238 @@
+use anyhow::Context;
+pub use client::*;
+use quinn::Connecting;
+pub use server::*;
+use tokio::sync::mpsc::Sender;
+
+use crate::io::write_message;
+
+use super::{read_message, ConnEvent, ConnHandle};
+
+pub const WAHT_ALPN: &[u8] = b"n0/waht/0";
+
+pub mod tls;
+
+pub async fn conn_actor(
+    conn: Connecting,
+    tx: Sender<ConnEvent>,
+    is_initiator: bool,
+) -> anyhow::Result<()> {
+    let conn = conn.await?;
+    let conn_id = conn.stable_id();
+
+    let (mut send, mut recv) = match is_initiator {
+        true => conn.open_bi().await?,
+        false => conn.accept_bi().await?,
+    };
```

Comment:
I mention this lower down as well.  Protocol-wise I think it would be nicer if each request was its own stream instead of having a single stream which handles all commands and responses.

Reply:
Yup, I think I agree. See also #1. But also depends on how much semantics from QUIC we take want to take into the protocol.